generate-package-notes.sh

```diff
@@ -46,11 +137,18 @@ write_script() {
     printf 'INSERT AFTER .note.gnu.build-id;\n'
 }
 
-# Not supported on every distro
-if [ ! -r /usr/lib/system-release-cpe ]; then
-    cpe="$(cat /usr/lib/system-release-cpe)"
-    json_cpe=",\"osCpe\": \"${cpe}\""
+if ! parse_options "$@" && [ "$#" -gt 0 ]; then
+    # Not supported on every distro
+    if [ -r /usr/lib/system-release-cpe ]; then
+        cpe="$(cat /usr/lib/system-release-cpe)"
+        json_cpe=",\"osCpe\":\"${cpe}\""
+    fi
+
+    # old-style invocation with positional parameters for backward compatibility
+    json="$(printf '{"type":"rpm","name":"%s","version":"%s","architecture":"%s"%s}' "$1" "$2" "$3" "$json_cpe")"
```

Comment:
FWIW, I don't think we need to worry about backwards compatibility here. This was never released, and I'll just adjust to the latest version in Fedora.